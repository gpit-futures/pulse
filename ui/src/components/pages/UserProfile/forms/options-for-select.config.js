const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
