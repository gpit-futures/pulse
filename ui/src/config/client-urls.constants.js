import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
