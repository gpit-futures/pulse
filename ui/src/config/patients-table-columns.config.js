import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
