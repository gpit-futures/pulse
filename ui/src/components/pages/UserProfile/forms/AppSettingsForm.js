import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
