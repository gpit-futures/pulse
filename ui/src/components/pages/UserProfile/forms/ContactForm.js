import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
