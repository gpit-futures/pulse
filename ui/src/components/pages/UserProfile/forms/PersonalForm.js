import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
