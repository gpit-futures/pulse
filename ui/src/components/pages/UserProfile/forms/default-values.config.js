import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
