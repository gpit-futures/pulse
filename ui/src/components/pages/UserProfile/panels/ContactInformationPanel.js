import React, { PureComponent } from 'react';

import ControlPanel from '../ControlPanel';
import ContactForm from '../forms/ContactForm';
import { valuesContactFormLabels } from '../forms.config';

const CONTACT_INFORMATION = 'contactInformation';

export default class ContactInformationPanel extends PureComponent {
  render() {
    const {
      user,
      openedPanel,
      editedPanel,
      onShow,
      onExpand,
      onEdit,
      onCancel,
      isShowControlPanel,
      isSaveButton,
    } = this.props;

    return (
      <ControlPanel
        name={CONTACT_INFORMATION}
        title="Contact Information"
        isOpen={openedPanel === CONTACT_INFORMATION}
        onShow={onShow}
        onExpand={onExpand}
        onEdit={onEdit}
        editedPanel={editedPanel}
        onCancel={onCancel}
        isShowControlPanel={isShowControlPanel}
        isSaveButton={isSaveButton}
      >
        <div>
          {editedPanel[CONTACT_INFORMATION] ?
            <ContactForm/> :
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div className="row">
                    <div className="col-xs-12 col-md-6">
                      <div className="row">
                        <div className="col-md-11">
                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.ADDRESS}</label>
                            <div className="form-control-static ng-binding">6801 Tellus Street</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.CITY}</label>
                            <div className="form-control-static ng-binding">Westmorland</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.STATE}</label>
                            <div className="form-control-static ng-binding">Westmorland</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.POSTAL_CODE}</label>
                            <div className="form-control-static ng-binding">Box 306</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.SELECT_COUNTRY}</label>
                            <div className="form-control-static ng-binding">USA</div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div className="col-xs-12 col-md-6">
                      <div className="row">
                        <div className="col-md-11 col-md-offset-1">
                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.PHONE}</label>
                            <div className="form-control-static ng-binding">07624 647524</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">{valuesContactFormLabels.EMAIL}</label>
                            <div className="form-control-static">{user.email}</div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          }
        </div>
      </ControlPanel>
    )
  }
}