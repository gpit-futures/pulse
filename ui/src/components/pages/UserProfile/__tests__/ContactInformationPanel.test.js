import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import ContactInformationPanel from '../panels/ContactInformationPanel';
import { valuesContactFormLabels } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

const CONTACT_INFORMATION = 'contactInformation';

const testProps = {
  user: {
    sub: '28AD8576-1948-4C84-8B5E-55FB7EE027CE',
    given_name: 'Bob',
    family_name: 'Smith',
    email: 'bob.smith@gmail.com',
    tenant: null,
    role: 'IDCR',
    roles: [
      'IDCR',
    ],
  },
  openedPanel: CONTACT_INFORMATION,
  editedPanel: {},
  onShow: () => {},
  onExpand: () => {},
  onEdit: () => {},
  onCancel: () => {},
  isShowControlPanel: true,
  isSaveButton: false,
};

describe('Component <ContactInformationPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <ContactInformationPanel
        user={testProps.user}
        openedPanel={testProps.openedPanel}
        editedPanel={testProps.editedPanel}
        onShow={testProps.onShow}
        onExpand={testProps.onExpand}
        onEdit={testProps.onEdit}
        onCancel={testProps.onCancel}
        isShowControlPanel={testProps.isShowControlPanel}
        isSaveButton={testProps.isSaveButton}
      />
    );

    expect(component.find('ControlPanel')).toHaveLength(1);
    expect(component.find('ControlPanel').props().title).toEqual('Contact Information');
    expect(component.find('ControlPanel').props().name).toEqual('contactInformation');

    expect(component.find('.control-label').at(0).text()).toEqual(valuesContactFormLabels.ADDRESS);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesContactFormLabels.CITY);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesContactFormLabels.STATE);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesContactFormLabels.POSTAL_CODE);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesContactFormLabels.SELECT_COUNTRY);
    expect(component.find('.control-label').at(5).text()).toEqual(valuesContactFormLabels.PHONE);
    expect(component.find('.control-label').at(6).text()).toEqual(valuesContactFormLabels.EMAIL);

    expect(component.find('.form-control-static').at(0).text()).toEqual('6801 Tellus Street');
    expect(component.find('.form-control-static').at(1).text()).toEqual('Westmorland');
    expect(component.find('.form-control-static').at(2).text()).toEqual('Westmorland');
    expect(component.find('.form-control-static').at(3).text()).toEqual('Box 306');
    expect(component.find('.form-control-static').at(4).text()).toEqual('USA');
    expect(component.find('.form-control-static').at(5).text()).toEqual('07624 647524');
    expect(component.find('.form-control-static').at(6).text()).toEqual(testProps.user.email);


    expect(component).toMatchSnapshot();

    component.setProps({ editedPanel: { contactInformation: true }});
    expect(component.find('ReduxForm')).toHaveLength(1);

  });
});

