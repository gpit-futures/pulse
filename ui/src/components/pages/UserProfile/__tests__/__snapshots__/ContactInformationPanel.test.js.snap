// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ContactInformationPanel /> should renders with all props correctly 1`] = `
<ControlPanel
  editedPanel={Object {}}
  isOpen={true}
  isSaveButton={false}
  isShowControlPanel={true}
  name="contactInformation"
  onCancel={[Function]}
  onEdit={[Function]}
  onExpand={[Function]}
  onShow={[Function]}
  title="Contact Information"
>
  <div>
    <div
      className="panel-body-inner"
    >
      <div
        className="form"
      >
        <div
          className="form-group-wrapper"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-12 col-md-6"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-11"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Address
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      6801 Tellus Street
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      City
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      Westmorland
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      State
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      Westmorland
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Postal Code
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      Box 306
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Country
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      USA
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-xs-12 col-md-6"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-11 col-md-offset-1"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Phone Number
                    </label>
                    <div
                      className="form-control-static ng-binding"
                    >
                      07624 647524
                    </div>
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      className="control-label"
                    >
                      Email
                    </label>
                    <div
                      className="form-control-static"
                    >
                      bob.smith@gmail.com
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</ControlPanel>
`;
